src/shared_modules/utils/rocksDBOptions.hpp

```diff
@@ -87,8 +87,11 @@ namespace Utils
             options.create_if_missing = true;
             // If true, log files will be kept around to restore the database
             options.keep_log_file_num = 1;
-            // Log level for the info log.
-            options.info_log_level = rocksdb::InfoLogLevel::FATAL_LEVEL;
+            // Setting INFO level for the info log. We'll have up to 10 files of 10MB each.
+            options.info_log_level = rocksdb::InfoLogLevel::INFO_LEVEL;
+            options.keep_log_file_num = 10;
+            options.max_log_file_size = 10 * 1024 * 1024;
+            options.recycle_log_file_num = 10;
```

Comment:
After further investigation, I've found that these configurations are related to the repair mechanism (and not "logging" itself)

If we read the ['Data Structures That Power Your Database'](https://github.com/user-attachments/files/16327005/Designing.Data-Intensive.Applications.pdf) section of the "Designing Data-Intensive Applications" book, we see a detailed explanation of how SST tables work, and finally a hint of how this repair mechanism works:

> if the database crashes, the most recent writes (which are in the memtable but not yet written out to disk) are lost. In order to avoid that problem, we can keep a separate log on disk to which every write is immediately appended, just like in the previous section. That log is not in sorted order, but that doesn’t matter, because its only purpose is to restore the memtable after a crash. Every time the memtable is written out to an SSTable, the corresponding log can be discarded.

> [!NOTE]
> This means that repairs will happen to the latest data, so the loss of information is minimal 😄 

And if we see the [rocksDB repairer documentation](https://github.com/facebook/rocksdb/wiki/RocksDB-Repairer#repair-process):
![image](https://github.com/user-attachments/assets/22ca72f5-2011-4028-a22d-7b697f5dcfed)

We see that again the logs are repair to the repair mechanism

Reply:
Hi @sebasfalcone 

That section of the book might be talking about the WAL (https://github.com/facebook/rocksdb/wiki/Terminology)

> Write-Ahead-Log (WAL) or log: A log file used to recover data that is not yet flushed to SST files, during DB recovery.

The options aren't well documented because the terms are confusing if you ask me.

In an example directory, we can see the following files

```
root@fb659c773250:/var/ossec# ls queue/keystore/
000010.sst  000021.log  CURRENT  IDENTITY  LOCK  LOG  LOG.old.1721707761172171  LOG.old.1721708086022965  LOG.old.1721708086044587  MANIFEST-000022  OPTIONS-000020  OPTIONS-000024
```

The files like `000021.log` are WALs, and those like `LOG.old.1721708086044587` are the real logging.
For example, here we have the content of a log file

<details><summary>Details</summary>
<p>

```
2024/07/23-04:14:46.024782 140163611096640        Options.write_buffer_size: 33554432
2024/07/23-04:14:46.024784 140163611096640  Options.max_write_buffer_number: 2
2024/07/23-04:14:46.024786 140163611096640          Options.compression: NoCompression
2024/07/23-04:14:46.024788 140163611096640                  Options.bottommost_compression: Disabled
2024/07/23-04:14:46.024790 140163611096640       Options.prefix_extractor: nullptr
2024/07/23-04:14:46.024791 140163611096640   Options.memtable_insert_with_hint_prefix_extractor: nullptr
2024/07/23-04:14:46.024793 140163611096640             Options.num_levels: 4
2024/07/23-04:14:46.024795 140163611096640        Options.min_write_buffer_number_to_merge: 1
2024/07/23-04:14:46.024802 140163611096640     Options.max_write_buffer_number_to_maintain: 0
2024/07/23-04:14:46.024804 140163611096640     Options.max_write_buffer_size_to_maintain: 0
2024/07/23-04:14:46.024806 140163611096640            Options.bottommost_compression_opts.window_bits: -14
2024/07/23-04:14:46.024808 140163611096640                  Options.bottommost_compression_opts.level: 32767
2024/07/23-04:14:46.024810 140163611096640               Options.bottommost_compression_opts.strategy: 0
2024/07/23-04:14:46.024812 140163611096640         Options.bottommost_compression_opts.max_dict_bytes: 0
2024/07/23-04:14:46.024814 140163611096640         Options.bottommost_compression_opts.zstd_max_train_bytes: 0
2024/07/23-04:14:46.024816 140163611096640         Options.bottommost_compression_opts.parallel_threads: 1
2024/07/23-04:14:46.024817 140163611096640                  Options.bottommost_compression_opts.enabled: false
2024/07/23-04:14:46.024819 140163611096640         Options.bottommost_compression_opts.max_dict_buffer_bytes: 0
2024/07/23-04:14:46.024821 140163611096640         Options.bottommost_compression_opts.use_zstd_dict_trainer: true
2024/07/23-04:14:46.024823 140163611096640            Options.compression_opts.window_bits: -14
2024/07/23-04:14:46.024825 140163611096640                  Options.compression_opts.level: 32767
2024/07/23-04:14:46.024827 140163611096640               Options.compression_opts.strategy: 0
2024/07/23-04:14:46.024829 140163611096640         Options.compression_opts.max_dict_bytes: 0
2024/07/23-04:14:46.024831 140163611096640         Options.compression_opts.zstd_max_train_bytes: 0
2024/07/23-04:14:46.024833 140163611096640         Options.compression_opts.use_zstd_dict_trainer: true
2024/07/23-04:14:46.024835 140163611096640         Options.compression_opts.parallel_threads: 1
2024/07/23-04:14:46.024838 140163611096640                  Options.compression_opts.enabled: false
2024/07/23-04:14:46.024843 140163611096640         Options.compression_opts.max_dict_buffer_bytes: 0
2024/07/23-04:14:46.024845 140163611096640      Options.level0_file_num_compaction_trigger: 4
2024/07/23-04:14:46.024847 140163611096640          Options.level0_slowdown_writes_trigger: 20
2024/07/23-04:14:46.024849 140163611096640              Options.level0_stop_writes_trigger: 36
2024/07/23-04:14:46.024850 140163611096640                   Options.target_file_size_base: 67108864
2024/07/23-04:14:46.024852 140163611096640             Options.target_file_size_multiplier: 1
2024/07/23-04:14:46.024854 140163611096640                Options.max_bytes_for_level_base: 268435456
2024/07/23-04:14:46.024856 140163611096640 Options.level_compaction_dynamic_level_bytes: 0
2024/07/23-04:14:46.024859 140163611096640          Options.max_bytes_for_level_multiplier: 10.000000
2024/07/23-04:14:46.024862 140163611096640 Options.max_bytes_for_level_multiplier_addtl[0]: 1
2024/07/23-04:14:46.024864 140163611096640 Options.max_bytes_for_level_multiplier_addtl[1]: 1
2024/07/23-04:14:46.024866 140163611096640 Options.max_bytes_for_level_multiplier_addtl[2]: 1
2024/07/23-04:14:46.024867 140163611096640 Options.max_bytes_for_level_multiplier_addtl[3]: 1
2024/07/23-04:14:46.024869 140163611096640 Options.max_bytes_for_level_multiplier_addtl[4]: 1
2024/07/23-04:14:46.024871 140163611096640 Options.max_bytes_for_level_multiplier_addtl[5]: 1
2024/07/23-04:14:46.024873 140163611096640 Options.max_bytes_for_level_multiplier_addtl[6]: 1
2024/07/23-04:14:46.024875 140163611096640       Options.max_sequential_skip_in_iterations: 8
2024/07/23-04:14:46.024877 140163611096640                    Options.max_compaction_bytes: 1677721600
2024/07/23-04:14:46.024879 140163611096640   Options.ignore_max_compaction_bytes_for_input: true
2024/07/23-04:14:46.024880 140163611096640                        Options.arena_block_size: 1048576
2024/07/23-04:14:46.024882 140163611096640   Options.soft_pending_compaction_bytes_limit: 68719476736
2024/07/23-04:14:46.024884 140163611096640   Options.hard_pending_compaction_bytes_limit: 274877906944
2024/07/23-04:14:46.024886 140163611096640                Options.disable_auto_compactions: 0
2024/07/23-04:14:46.024889 140163611096640                        Options.compaction_style: kCompactionStyleLevel
2024/07/23-04:14:46.024891 140163611096640                          Options.compaction_pri: kMinOverlappingRatio
2024/07/23-04:14:46.024893 140163611096640 Options.compaction_options_universal.size_ratio: 1
2024/07/23-04:14:46.024895 140163611096640 Options.compaction_options_universal.min_merge_width: 2
2024/07/23-04:14:46.024897 140163611096640 Options.compaction_options_universal.max_merge_width: 4294967295
2024/07/23-04:14:46.024899 140163611096640 Options.compaction_options_universal.max_size_amplification_percent: 200
2024/07/23-04:14:46.024901 140163611096640 Options.compaction_options_universal.compression_size_percent: -1
2024/07/23-04:14:46.024903 140163611096640 Options.compaction_options_universal.stop_style: kCompactionStopStyleTotalSize
2024/07/23-04:14:46.024904 140163611096640 Options.compaction_options_fifo.max_table_files_size: 1073741824
2024/07/23-04:14:46.024906 140163611096640 Options.compaction_options_fifo.allow_compaction: 0
2024/07/23-04:14:46.024911 140163611096640                   Options.table_properties_collectors: 
2024/07/23-04:14:46.024919 140163611096640                   Options.inplace_update_support: 0
2024/07/23-04:14:46.024921 140163611096640                 Options.inplace_update_num_locks: 10000
2024/07/23-04:14:46.024923 140163611096640               Options.memtable_prefix_bloom_size_ratio: 0.000000
2024/07/23-04:14:46.024926 140163611096640               Options.memtable_whole_key_filtering: 0
2024/07/23-04:14:46.024927 140163611096640   Options.memtable_huge_page_size: 0
2024/07/23-04:14:46.024929 140163611096640                           Options.bloom_locality: 0
2024/07/23-04:14:46.024931 140163611096640                    Options.max_successive_merges: 0
2024/07/23-04:14:46.024938 140163611096640                Options.optimize_filters_for_hits: 0
2024/07/23-04:14:46.024939 140163611096640                Options.paranoid_file_checks: 0
2024/07/23-04:14:46.024943 140163611096640                Options.force_consistency_checks: 1
2024/07/23-04:14:46.024946 140163611096640                Options.report_bg_io_stats: 0
2024/07/23-04:14:46.024948 140163611096640                               Options.ttl: 2592000
2024/07/23-04:14:46.024950 140163611096640          Options.periodic_compaction_seconds: 0
2024/07/23-04:14:46.024952 140163611096640  Options.preclude_last_level_data_seconds: 0
2024/07/23-04:14:46.024953 140163611096640    Options.preserve_internal_time_seconds: 0
2024/07/23-04:14:46.024955 140163611096640                       Options.enable_blob_files: false
2024/07/23-04:14:46.024957 140163611096640                           Options.min_blob_size: 0
2024/07/23-04:14:46.024959 140163611096640                          Options.blob_file_size: 268435456
2024/07/23-04:14:46.024961 140163611096640                   Options.blob_compression_type: NoCompression
2024/07/23-04:14:46.024963 140163611096640          Options.enable_blob_garbage_collection: false
2024/07/23-04:14:46.024965 140163611096640      Options.blob_garbage_collection_age_cutoff: 0.250000
2024/07/23-04:14:46.024967 140163611096640 Options.blob_garbage_collection_force_threshold: 1.000000
2024/07/23-04:14:46.024970 140163611096640          Options.blob_compaction_readahead_size: 0
2024/07/23-04:14:46.024971 140163611096640                Options.blob_file_starting_level: 0
2024/07/23-04:14:46.024974 140163611096640 Options.experimental_mempurge_threshold: 0.000000
2024/07/23-04:14:46.034846 140163611096640 [db/version_set.cc:5876] Recovered from manifest file:queue/keystore/MANIFEST-000014 succeeded,manifest_file_number is 14, next_file_number is 16, last_sequence is 1, log_number is 5,prev_log_number is 0,max_column_family is 1,min_log_number_to_keep is 5
2024/07/23-04:14:46.034892 140163611096640 [db/version_set.cc:5891] Column family [default] (ID 0), log number is 5
2024/07/23-04:14:46.034896 140163611096640 [db/version_set.cc:5891] Column family [indexer] (ID 1), log number is 5
2024/07/23-04:14:46.035093 140163611096640 [db/db_impl/db_impl_open.cc:636] DB ID: 6506c4e0-a706-4141-a208-950fb9fc5aa1
2024/07/23-04:14:46.035600 140163611096640 EVENT_LOG_v1 {"time_micros": 1721708086035581, "job": 1, "event": "recovery_started", "wal_files": [13]}
2024/07/23-04:14:46.035609 140163611096640 [db/db_impl/db_impl_open.cc:1129] Recovering log #13 mode 2
2024/07/23-04:14:46.035824 140163611096640 EVENT_LOG_v1 {"time_micros": 1721708086035821, "job": 1, "event": "recovery_finished"}
2024/07/23-04:14:46.036111 140163611096640 [db/version_set.cc:5346] Creating manifest 18
2024/07/23-04:14:46.042967 140163611096640 [file/delete_scheduler.cc:73] Deleted file queue/keystore/000013.log immediately, rate_bytes_per_sec 0, total_trash_size 0 max_trash_db_ratio 0.250000
2024/07/23-04:14:46.043006 140163611096640 [db/db_impl/db_impl_open.cc:2084] SstFileManager instance 0x7f7a5e709e00
2024/07/23-04:14:46.043078 140163611096640 DB pointer 0x7f7a5e852800
2024/07/23-04:14:46.043529 140163611096640 [db/db_impl/db_impl.cc:495] Shutdown: canceling all background work
2024/07/23-04:14:46.043900 140163611096640 [db/db_impl/db_impl.cc:698] Shutdown complete
```

</p>
</details>